### ISSUE TITLE
🐛 Vue v-on directives should be able to contain multiple statements

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#10519]
### Environment information

<details>

```bash

playground

```

</details>

### What happened?

https://biomejs.dev/playground/?analyzerFixMode=safeAndUnsafeFixes&tab=formatter&pane=Diagnostics&code=PABOAGEAdgBDAHIAZQBhAHQAZQBSAG8AbwBtAAoAIAAgACAAIABAAGMAcgBlAGEAdABlAHQAZQBtAHAAPQAiAAoAIAAgACAAIAAgACAAIAAgAGQAcgBhAHcAZQByACAAPQAgAGYAYQBsAHMAZQA7AAoAIAAgACAAIAAgACAAIAAgAGMAcgBlAGEAdABlAFQAZQBtAHAAUgBvAG8AbQAoACkAOwAKACAAIAAgACAAIgAKACAAIAAgACAAQABjAHIAZQBhAHQAZQBwAGUAcgBtAD0AIgAKACAAIAAgACAAIAAgACAAIABkAHIAYQB3AGUAcgAgAD0AIABmAGEAbABzAGUAOwAKACAAIAAgACAAIAAgACAAIABzAGgAbwB3AEMAcgBlAGEAdABlAFIAbwBvAG0ARgBvAHIAbQAgAD0AIAB0AHIAdQBlADsACgAgACAAIAAgACIACgAvAD4A&language=vue

### Expected result

It should not throw a parsing error, as this is valid vue syntax.

Right now, IIRC, we just accept AnyJsExpression here. I'll make a vue playground/find docs to make sure we nail the exact behavior this time.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
vue playground (2 statements): 
```vue
<button @click="counter++; counter++;">Add</button>
```

https://play.vuejs.org/#eNp9UkFOwzAQ/MrKJ1CrBASnEioK6gEOgKBHX9xk26Z1bMtel0hR/o6dNCkHVMmy1jPj0azXDVsYkxw9shnLXG5LQ3OusDbaEhS4EV4SNFwBFILE1XWsOQFYJG9VzwBwyrVXhHYGNxHh1EZZ2MLK0tE3HAgrIwVhPFG29kRawVMuy/zwyNnJZjJ5gHPJ2XxRFHCbpb28v2rmqx3CyUCs9RFhJxysERV0dlhA0ww20LZAZYUuyVITDLJ0DMKmjFyu1abcJnunVXiJrq2YpjKlRPthqNTKcTYbG2ZCSv3z1mFkPU4HPN9hfvgH37s6Ypx9WnRoj8jZyJGwW6SeXn6/Yx3qkax04WVQXyC/0GnpY8Ze9uxVEWL/0XVpX6s41VJtV25ZEyo3NBWDRmXb6TkL3+HlQuvnuHfJfXcvzJm1vyVexU8=

however, vue does not really like this:

```vue
<button @click="() => {counter++}; counter++;">Add</button>
```

https://play.vuejs.org/#eNp9UtFOwyAU/ZUbnjQurUaftq5xmj3ogxrdIy8dvdu6USBwmU2a/rvQbtUHs4SQyzmHw7lAyxbGJEePbMoyJ2xlKOcKG6MtQYmbwkuCliuAsqDi6jrWnAAskrdqYAA4Ce0VoZ3CbUQ4dVEWpjCydPQNC8LayIIwrihbeyKt4FHIShzmnIUD5jm0J7ebm24GYz3jLF+UJdxl6bBtsDD5aodwMirW+oiwKxysERX0tlhC255toOuAqhpdkqUmGGTpGIhNGDmh1abaJnunVbiRvj3OhK5NJdG+G6q0cpxNx8ZZIaX+fu0xsh4nZ1zsUBz+wfeuiRhnHxYd2iNyNnJU2C3SQC+/3rAJ9UjWuvQyqC+Qn+i09DHjIHvyqgyx/+j6tC91fN1KbVdu2RAqd24qBo3KrtdzFr7F84XWf+PeJw/9vvDerPsBQW3HUw==

It appears that Vue has some hueristics for this behavior: https://github.com/vuejs/core/blob/86ad0764fd9f7b01cef75b4fc941b03419306bf8/packages/compiler-core/src/transforms/vOn.ts#L79-L92

Related unit tests:
- https://github.com/vuejs/core/blob/86ad0764fd9f7b01cef75b4fc941b03419306bf8/packages/compiler-core/__tests__/transforms/vOn.spec.ts
